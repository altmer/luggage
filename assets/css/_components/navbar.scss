.nav-link {
  cursor: pointer;
}

.nav-item {
  .dropdown-item {
    cursor: pointer;
    padding-bottom: 7px;
    padding-top: 7px;
  }
}

.navbar-toggleable-xl .navbar-nav .dropdown-menu {
  position: absolute;
}

.navbar {
  .search-opener {
    img {
      margin-right: 1ch;
      opacity: .6;
    }

    &:focus {
      outline: none;
    }

    &:hover {
      img {
        opacity: 1;
      }
    }
  }
}
