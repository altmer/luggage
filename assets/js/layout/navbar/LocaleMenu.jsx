import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { UncontrolledNavDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';

export class LocaleMenu extends Component {
  constructor(props) {
    super(props);
    this.handleClickEn = this.props.setLocale.bind(null, 'en');
    this.handleClickDe = this.props.setLocale.bind(null, 'de');
    this.handleClickRu = this.props.setLocale.bind(null, 'ru');
  }

  render() {
    return (
      <UncontrolledNavDropdown className="locale-menu">
        <DropdownToggle caret className="nav-link" tag="a">
          {this.props.locale.toUpperCase()}
        </DropdownToggle>
        <DropdownMenu right>
          <DropdownItem className="locale-link-en" onClick={this.handleClickEn}>
            EN
          </DropdownItem>
          <DropdownItem className="locale-link-de" onClick={this.handleClickDe}>
            DE
          </DropdownItem>
          <DropdownItem className="locale-link-ru" onClick={this.handleClickRu}>
            RU
          </DropdownItem>
        </DropdownMenu>
      </UncontrolledNavDropdown>
    );
  }
}

LocaleMenu.propTypes = {
  locale: PropTypes.string.isRequired,
  setLocale: PropTypes.func.isRequired,
};

export default LocaleMenu;
