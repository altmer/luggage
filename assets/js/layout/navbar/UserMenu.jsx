import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';
import { UncontrolledNavDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
import { Translate } from 'react-i18nify';

import Avatar from '../../users/components/Avatar';

export const UserMenu = props => (
  <UncontrolledNavDropdown className="user-menu">
    <DropdownToggle className="nav-link" tag="a">
      <Avatar {...props.user} />
    </DropdownToggle>
    <DropdownMenu right>
      <DropdownItem className="user-link" tag={Link} to={`/users/${props.user.id}/show`} >
        <Translate value="users.myProfile" />
      </DropdownItem>
      <DropdownItem className="user-link" tag={Link} to="/users/edit">
        <Translate value="users.changePassword" />
      </DropdownItem>
      <DropdownItem divider />
      <DropdownItem className="user-link" onClick={props.signOut}>
        <Translate value="sessions.logout" />
      </DropdownItem>
    </DropdownMenu>
  </UncontrolledNavDropdown>
);

UserMenu.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number,
    avatar: PropTypes.shape({
      thumb: PropTypes.string,
    }),
    initials: PropTypes.string,
    color: PropTypes.string,
  }),
  signOut: PropTypes.func.isRequired,
};

UserMenu.defaultProps = {
  user: {
    avatar: {
      thumb: null,
    },
  },
};

export default UserMenu;
