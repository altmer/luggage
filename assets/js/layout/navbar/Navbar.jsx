import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Translate } from 'react-i18nify';
import axios from 'axios';

import LocaleMenu from '../navbar/LocaleMenu';
import UserMenu from '../navbar/UserMenu';
import { signOut, setLocale } from '../../auth/sessions/ducks';

const updateLocale = locale => (
  axios.put('/api/sessions/locale', { locale })
);

export class Navbar extends Component {
  constructor(props) {
    super(props);
    this.setLocale = this.setLocale.bind(this);
  }

  setLocale(locale) {
    updateLocale(locale);
    this.props.setLocale(locale);
  }

  userMenu() {
    const { currentUser } = this.props;
    if (currentUser) {
      return (
        <UserMenu
          user={currentUser}
          signOut={this.props.signOut}
        />
      );
    }
    return null;
  }

  login() {
    const { currentUser } = this.props;
    if (!currentUser) {
      return (<li className="nav-item">
        <Link to="/sign_in" className="nav-link">
          <Translate value="sessions.login" />
        </Link>
      </li>);
    }
    return null;
  }

  register() {
    const { currentUser } = this.props;
    if (!currentUser) {
      return (<li className="nav-item hidden-xs-down">
        <Link to="/sign_up" className="nav-link">
          <Translate value="sessions.register" />
        </Link>
      </li>);
    }
    return null;
  }

  render() {
    return (
      <div className="container">
        <nav className="navbar navbar-light navbar-toggleable-xl my-2">
          <Link className="navbar-brand" to="/">
            The Luggage
          </Link>
          <ul className="navbar-nav mr-auto">
            <li className="nav-item">
              <Link to="/search" className="search-opener nav-link">
                <img src="/images/search.svg" alt="search" />
                <span className="hidden-xs-down">
                  <Translate value="common.search" />
                </span>
              </Link>
            </li>
          </ul>
          <ul className="navbar-nav">
            <LocaleMenu locale={this.props.locale} setLocale={this.setLocale} />
            {this.register()}
            {this.login()}
            {this.userMenu()}
          </ul>
        </nav>
      </div>
    );
  }
}

Navbar.propTypes = {
  locale: PropTypes.string.isRequired,
  currentUser: PropTypes.shape({
    name: PropTypes.string,
  }),
  signOut: PropTypes.func.isRequired,
  setLocale: PropTypes.func.isRequired,
};

Navbar.defaultProps = {
  currentUser: undefined,
};

const mapStateToProps = state => (
  {
    currentUser: state.session.currentUser,
    locale: state.session.locale,
  }
);

const mapDispatchToProps = dispatch => (
  {
    signOut() {
      dispatch(signOut());
    },
    setLocale(locale) {
      dispatch(setLocale(locale));
    },
  }
);


export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
