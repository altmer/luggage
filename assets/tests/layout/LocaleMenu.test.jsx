import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { LocaleMenu } from '../../js/layout/navbar/LocaleMenu';

test('it renders locale menu', () => {
  const subject = renderer.create(
    <LocaleMenu
      locale="ru"
      setLocale={jest.fn()}
    />);
  expect(subject.toJSON()).toMatchSnapshot();
});

test('it calls setLocale callback on click', () => {
  const setLocaleMock = jest.fn();
  const subject = mount(
    <LocaleMenu
      locale="ru"
      setLocale={setLocaleMock}
    />);
  subject.find('.locale-link-en').simulate('click');
  expect(setLocaleMock).toHaveBeenCalledWith('en', expect.anything());
});
