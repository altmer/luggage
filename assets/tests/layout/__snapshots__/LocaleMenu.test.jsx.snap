// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders locale menu 1`] = `
<li
  className="locale-menu nav-item dropdown"
>
  <a
    aria-expanded={false}
    aria-haspopup="true"
    className="nav-link dropdown-toggle"
    color="secondary"
    data-toggle="dropdown"
    onClick={[Function]}
  >
    RU
  </a>
  <div
    aria-hidden={true}
    className="dropdown-menu dropdown-menu-right"
    role="menu"
    tabIndex="-1"
  >
    <button
      className="locale-link-en dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      EN
    </button>
    <button
      className="locale-link-de dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      DE
    </button>
    <button
      className="locale-link-ru dropdown-item"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      RU
    </button>
  </div>
</li>
`;
